### PR TITLE
IBM Cloud PostgreSQL

### PR DESCRIPTION
**Pull request by SubratoP1**

### CODE REVIEW

File: filter-plugin/logstash-filter-postgres-ibmcloud-guardium/Postgres.conf [hunk 1]
```diff
@@ -0,0 +1,82 @@
+//
+// Copyright IBM Corp. 2023 All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+input {
+  kafka{
+    #Insert "bootstrap_endpoints" string from the service credential in the Event stream.
```
[the remark this answers is not in the document]
Comment (SubratoP1) [reply filter-plugin/logstash-filter-postgres-ibmcloud-guardium/Postgres.conf:7]: Hi Pratiksha,

As I have mentioned user need to insert the "bootstrap_endpoints" string as it is which is inside the service credentials of Event-stream. Sample URLs here doesn't make any sense that's why we are using Place Holders.

**Review by PratikshaSonawane — APPROVED** [review]

File: filter-plugin/logstash-filter-postgres-ibmcloud-guardium/README.md
Comment (apurva-birajdar) [inline filter-plugin/logstash-filter-postgres-ibmcloud-guardium/README.md:1]: README format should be same to other README files. 
I see missing point **Meet-Postgres-IBMCloud**
example here : https://github.com/IBM/universal-connectors/blob/main/filter-plugin/logstash-filter-aurora-mysql-guardium/README.md#meet-aurora-mysql 

Please add.

Comment (apurva-birajdar) [inline filter-plugin/logstash-filter-postgres-ibmcloud-guardium/README.md:1]: @miriam.lezak@ibm.com 
Requesting you to please review the documentation part.

File: filter-plugin/logstash-filter-postgres-ibmcloud-guardium/build.gradle
Comment (apurva-birajdar) [inline filter-plugin/logstash-filter-postgres-ibmcloud-guardium/build.gradle:1]: @SubratoP1 
have we have done the WICKED CLI scan for all open source libraries used here ?
Hope we got the approvals on the same already.

**Review by apurva-birajdar — COMMENTED** [review]
@SubratoP1 
can we please add sample audit log file please ?

File: filter-plugin/logstash-filter-postgres-ibmcloud-guardium/build.gradle
Comment (SubratoP1) [reply filter-plugin/logstash-filter-postgres-ibmcloud-guardium/build.gradle:1]: Hi Apurva,

WICKED CLI scan report already shared. Please check and verify.

File: filter-plugin/logstash-filter-postgres-ibmcloud-guardium/IcdPostgresOverKafkaEvent/filter.conf
Comment (apurva-birajdar) [inline filter-plugin/logstash-filter-postgres-ibmcloud-guardium/IcdPostgresOverKafkaEvent/filter.conf:1]: Copyright statement should be 2024 not 2023.

**Review by apurva-birajdar — COMMENTED** [review]
https://github.com/SubratoP1/universal-connectors/tree/IbmPostgres/filter-plugin/logstash-filter-postgres-ibmcloud-guardium/gradle/wrapper/wrapper/dists/gradle-7.1.1-bin/f29rtwfnc96ub43tt7p47zsru
 
Which is this extra directory ? Can we remove it.

File: filter-plugin/logstash-filter-postgres-ibmcloud-guardium/README.md
Comment (apurva-birajdar) [reply filter-plugin/logstash-filter-postgres-ibmcloud-guardium/README.md:1]: it is added in latest commit.

**Review by apurva-birajdar — APPROVED** [review]

File: filter-plugin/logstash-filter-postgres-ibmcloud-guardium/README.md [hunk 2]
```diff
@@ -1,4 +1,12 @@
 # IBM Cloud PostgresSQL-Guardium Logstash filter plug-in
+
+### Meet IBM PostgreSQL
+* Tested versions: v15
+* Environment: IBM CLOUD
+* Supported inputs: Kafka (pull)
+* Supported Guardium versions:
+  * Guardium Data Protection: 11.4 and above
```
Comment (mllezak) [inline filter-plugin/logstash-filter-postgres-ibmcloud-guardium/README.md:8]: 11.4 and later

File: filter-plugin/logstash-filter-postgres-ibmcloud-guardium/README.md [hunk 3]
```diff
@@ -1,4 +1,12 @@
...
+  
 This is a [Logstash](https://github.com/elastic/logstash) filter plug-in for the universal connector that is featured in IBM Security Guardium. It parses events and messages from the IBM Cloud PostgresSQL audit log into a [Guardium record](https://github.com/IBM/universal-connectors/blob/main/common/src/main/java/com/ibm/guardium/universalconnector/commons/structures/Record.java) instance (which is a standard structure made out of several parts). The information is then sent over to Guardium. Guardium records include the accessor (the person who tried to access the data), the session, data, and exceptions. If there are no errors, the data contains SQL commands are not parsed by this plug-in but rather forwarded as it is to Guardium to do the SQL parsing.
 
 The plug-in is free and open-source (Apache 2.0). It can be used as a starting point to develop additional filter plug-ins for the Guardium universal connector.
```
Comment (mllezak) [inline filter-plugin/logstash-filter-postgres-ibmcloud-guardium/README.md:12]: The plug-in is free and open-source (Apache 2.0). You can use it as a starting point to develop additional filter plug-ins for the Guardium universal connector.

**Review by mllezak — COMMENTED** [review]
For some reason, I couldn't edit or even add comments to the main text. However, it needed a lot of tweaking -- nothing major, but I did a bunch of cleanup -- Here:

[PostgreSQL_UC_README.md](https://github.com/IBM/universal-connectors/files/14129509/PostgreSQL_UC_README.md)

File: filter-plugin/logstash-filter-postgres-ibmcloud-guardium/README.md (hunk 2 above)
Comment (SubratoP1) [reply filter-plugin/logstash-filter-postgres-ibmcloud-guardium/README.md]: Hi mllezak,

Changes has been made in README according to your suggestion please have a look and verify.

File: filter-plugin/logstash-filter-postgres-ibmcloud-guardium/README.md [hunk 4]
```diff
@@ -1,4 +1,12 @@
 # IBM Cloud PostgresSQL-Guardium Logstash filter plug-in
+
+### Meet IBM PostgreSQL
+* Tested versions: v15
+* Environment: IBM CLOUD
+* Supported inputs: Kafka (pull)
+* Supported Guardium versions:
+  * Guardium Data Protection: 11.4 and above
+  
 This is a [Logstash](https://github.com/elastic/logstash) filter plug-in for the universal connector that is featured in IBM Security Guardium. It parses events and messages from the IBM Cloud PostgresSQL audit log into a [Guardium record](https://github.com/IBM/universal-connectors/blob/main/common/src/main/java/com/ibm/guardium/universalconnector/commons/structures/Record.java) instance (which is a standard structure made out of several parts). The information is then sent over to Guardium. Guardium records include the accessor (the person who tried to access the data), the session, data, and exceptions. If there are no errors, the data contains SQL commands are not parsed by this plug-in but rather forwarded as it is to Guardium to do the SQL parsing.
```
[the remark this answers is not in the document]
Comment (SubratoP1) [reply filter-plugin/logstash-filter-postgres-ibmcloud-guardium/README.md]: Hi mllezak,

Changes has been made in README according to your suggestion please have a look and verify.

File: filter-plugin/logstash-filter-postgres-ibmcloud-guardium/README.md (hunk 3 above)
Comment (SubratoP1) [reply filter-plugin/logstash-filter-postgres-ibmcloud-guardium/README.md]: Hi mllezak,

Changes has been made in README according to your suggestion please have a look and verify.

**Review by mllezak — APPROVED** [review]
Thank you @SubratoP1!

**Pull request merged**